pkgs/applications/editors/emacs/generic.nix

```diff
@@ -67,18 +67,18 @@ in stdenv.mkDerivation {
 
     # Make native compilation work both inside and outside of nix build
     (lib.optionalString nativeComp (let
-      libPath = lib.concatStringsSep ":" [
-        "${lib.getLib libgccjit}/lib/gcc/${targetPlatform.config}/${libgccjit.version}"
-        "${lib.getLib stdenv.cc.cc}/lib"
-        "${lib.getLib stdenv.glibc}/lib"
-      ];
+      libPath = (lib.concatStringsSep " "
+        (builtins.map (x: ''\"-L${x}\"'') [
+          "${lib.getLib libgccjit}/lib"
+          "${lib.getLib libgccjit}/lib/gcc/${targetPlatform.config}/${libgccjit.version}"
+          "${lib.getLib stdenv.cc.cc}/lib"
```

Comment:
I'm not sure `"${lib.getLib stdenv.cc.cc}/lib"` is necessary, I don't know why I had this in LIBRARY_PATH originally. Tested without and seems fine, but also harmless to leave in.

Reply:
I think -B is supposed to be recursive, previously when passing the library_path as an environmental variable to comp-async-workers (which never worked, but worked if you wrapped emacs) we had to pass subdirectories manually. It's unclear to me if this is still necessary (I lean towards no)

Reply:
Welp, I did a (not so quick) test run just now, and it seems we do need that specific directory - but we do not need the targetPlatform/version games. The others, on the other hand - can mostly go.

Reply:
Yes, you seem to be right -- I don't think we need the machine version/target platform based on the manual: 

```
-Bprefix
This option specifies where to find the executables, libraries, include files, and data files of the compiler itself.

The compiler driver program runs one or more of the subprograms cpp, cc1, as and ld. *It tries prefix as a prefix for each program it tries to run, both with and without ‘machine/version/’ for the corresponding target machine and compiler version.*
```

Reply:
I think we still need three directories: 

```
 "${lib.getLib libgccjit}/lib"
 "${lib.getLib libgccjit}/lib/gcc"
 "${lib.getLib stdenv.cc.libc}/lib"
```

Reply:
Yup - that's the ones I have now, plus the bin dirs from the platform cc & bintools. I just got that working with an unwrapped emacs binary, so I think that change I just pushed should be it.